tap_zendesk/__init__.py

```diff
@@ -199,7 +199,7 @@ def main():
         LOGGER.error("""No suitable authentication keys provided.""")
 
```

Comment:
@prijendev  How about create one new function and check access before this do_discover is called? In this case, we won't need to modify other functions.